src/ln/channelmanager.rs

```diff
@@ -2776,6 +2792,17 @@ impl ChannelMessageHandler for ChannelManager {
 		handle_error!(self, self.internal_announcement_signatures(their_node_id, msg), their_node_id)
 	}
 
+	fn handle_channel_update(&self, msg: &msgs::ChannelUpdate) -> Result<(), HandleError> {
+		let mut channel_state_lock = self.channel_state.lock().unwrap();
```

Comment:
While technically I think its safe without it, probably best to keep consistency and add a total_consistency_lock.read().unwrap() at the top.

Reply:
Added